examples/RNOneSignal/App.js

```diff
@@ -8,484 +8,700 @@
+                    returnKeyType="done"
+                    textAlign="center"
+                    placeholderTextColor="#d1dde3"
+                    editable={isEditable}
+                    autoCapitalize="none"
+                    onChangeText={callback}
+                />
+            </KeyboardAvoidingView>
+        );
+    }
+
+    /**
+     ADD MORE GENERIC UI METHODS HERE...
+     */
+
+    /**
+     Create the fields for displaying information about the demo app and some instruction
+        for modifying the demo app or react-native SDK
+     */
+    createTitleFields() {
+        // States used through-out the title fields
+        const {
+            debugText
+        } = this.state;
+
+        return (
+            <View style={styles.container}>
+                <View>
+                    <Image style={styles.imageStyle} source={{uri: imageUri}} />
+                </View>
+
+                <Text style={styles.welcome}>
+                    Welcome to React Native!
+                </Text>
+
+                <Text style={styles.instructions}>
+                    To get started using the OneSignal SDK, edit App.js
+                </Text>
+
+                <Text style={styles.instructions}>
+                    To get started modifying the OneSignal SDK, edit index.js
+                </Text>
+
+                <Text style={styles.instructions}>
+                    Double tap R on your keyboard to reload,{'\n'}
+                    Shake or press menu button for dev menu
+                </Text>
+
+                <Text style={styles.jsonDebugLabelText}>
+                    {debugText}
+                </Text>
+
+                <Text style={styles.instructions}>
+                    App Id:{'\n'}
+                    {appId}
+                </Text>
             </View>
-          </View>
-          <View style={{flexDirection: 'row', overflow: 'hidden'}}>
-            <View style={styles.buttonContainer}>
-              <Button
-                style={styles.button}
-                disabled={!this.state.emailEnabled}
-                onPress={() => {
-                  this.setState({
-                    animatingEmailButton: true,
-                    activityWidth: 20,
-                    activityMargin: 10,
-                  });
-
-                  OneSignal.setEmail(this.state.email, error => {
-                    console.log('Sent email with error: ', error);
-
-                    this.setState({
-                      animatingEmailButton: false,
-                      activityWidth: 0,
-                      activityMargin: 0,
+        );
+    }
+
+    /**
+     Create the fields necessary to test subscription with OneSignal SDK
+     */
+    createSubscribeFields() {
+        // States used through-out the subscription fields
+        const {
+            isSubscribed,
+            isSubscriptionLoading,
+            isPrivacyConsentLoading
+        } = this.state;
+
+        let table = [];
```

Comment:
nit, var naming, should this be `elements` instead of `table`?
Few other functions where we call this `table` too.

Reply:
I just think of it as a table of elements being returned to the render method